src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -138,6 +142,7 @@ public void stop() throws ConnectException {
     log.info("Stopping ElasticsearchSinkTask.");
     if (writer != null) {
       writer.stop();
+      writer = null;
```

Comment:
if `flush` and `stop` are executed in two different threads (which I think they could) this won't necessarily be visible without the variable being `volatile`